src/logic/tokens/store/actions/fetchTokens.js

```diff
@@ -37,25 +39,102 @@ const createERC721TokenContract = async () => {
   return erc721Token
 }
 
+const OnlyBalanceToken = {
+  contractName: 'OnlyBalanceToken',
+  abi: [
+    {
+      constant: true,
+      inputs: [
+        {
+          name: 'owner',
+          type: 'address',
+        },
+      ],
+      name: 'balanceOf',
+      outputs: [
+        {
+          name: '',
+          type: 'uint256',
+        },
+      ],
+      payable: false,
+      stateMutability: 'view',
+      type: 'function',
+    },
+  ],
+}
+
+// For the `batchRequest` of balances, we're just using the `balanceOf` method call.
+// So having a simple ABI only with `balanceOf` prevents errors
+// when instantiating non-standard ERC-20 Tokens.
+const createOnlyBalanceToken = async () => {
+  const web3 = getWeb3()
+  const token = await contract(OnlyBalanceToken)
+  token.setProvider(web3.currentProvider)
+  return token
+}
+
 export const getHumanFriendlyToken = ensureOnce(createHumanFriendlyTokenContract)
 
 export const getStandardTokenContract = ensureOnce(createStandardTokenContract)
 
 export const getERC721TokenContract = ensureOnce(createERC721TokenContract)
 
+export const getOnlyBalanceToken = ensureOnce(createOnlyBalanceToken)
+
 export const containsMethodByHash = async (contractAddress: string, methodHash: string) => {
   const web3 = getWeb3()
   const byteCode = await web3.eth.getCode(contractAddress)
 
   return byteCode.indexOf(methodHash.replace('0x', '')) !== -1
 }
 
-export const fetchTokens = () => async (dispatch: ReduxDispatch<GlobalState>) => {
+export const getTokenInstance = async (tokenAddress: string) => {
+  if (!tokenAddress) {
+    return null
+  }
+  const { tokens } = store.getState()
+  let tokenInstance = tokens.get(tokenAddress)
+  // If the token is inside the cache we return the cached token
+  if (tokenInstance) {
+    return tokenInstance
+  }
+  // Otherwise we fetch it, save it to the cache and return it
+  const tokenContract = await getHumanFriendlyToken()
+  tokenInstance = await tokenContract.at(tokenAddress)
+  const [tokenSymbol, tokenDecimals, name] = await Promise.all([
+    tokenInstance.symbol(),
+    tokenInstance.decimals(),
+    tokenInstance.name(),
+  ])
+  tokenInstance.symbol = tokenSymbol
+  tokenInstance.tokenDecimals = tokenDecimals
+  const savedToken = makeToken({
+    address: tokenAddress,
+    name: name ? name : tokenSymbol,
+    symbol: tokenSymbol,
+    decimals: tokenDecimals,
+    logoUri: '',
+  })
+  const newTokens = tokens.set(tokenAddress, savedToken)
+  const dispatch = store.dispatch
+  dispatch(saveTokens(newTokens))
```

Comment:
I think it's fine to call it like `store.dispatch` like you do with getState on line 96 :)